phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexTool.java

```diff
@@ -682,9 +694,15 @@ public int run(String[] args) throws Exception {
                 if (htable != null) {
                     htable.close();
                 }
-                if(regionLocator != null) {
+                if (hConn != null) {
+                    hConn.close();
```

Comment:
same here, if any of these closes throw, the ones afterwards won't get closed

Reply:
Might want to take a look at Guava's Closer to see if that can help with these cases.  Register all the closeables, then call closer.close() once